config.go

```diff
@@ -42,293 +40,98 @@ const (
-
-	// Files to load without interpolation.
-	staticFiles []string
-
-	// Dirs to load from.
-	dirs []string
-
-	// Where to look for environment variables.
-	lookUp lookUpFunc
-}
-
-// DefaultLoader is going to be used by a service if config is not specified.
-// First values are going to be looked in dynamic providers, then in command line provider
-// and YAML provider is going to be the last.
-var DefaultLoader = NewLoader(commandLineProviderFunc)
-
-// NewLoader returns a default Loader with providers overriding the YAML provider.
-func NewLoader(providers ...ProviderFunc) *Loader {
-	l := &Loader{
-		envPrefix: "APP",
-		dirs:      []string{".", "./config"},
-		lookUp:    os.LookupEnv,
-	}
-
-	// Order is important: we want users to be able to override static provider
-	l.RegisterProviders(l.YamlProvider())
-	l.RegisterProviders(providers...)
-
-	return l
-}
-
-// TestConfig is Provider that can be used for testing. It loads configuration from
-// base.yaml and test.yaml files.
-var TestConfig = func() Provider {
-	l := NewLoader()
-	l.SetConfigFiles(_baseFile, "test.yaml")
-	l.SetDirs(func() []string {
-		dirs := []string{}
-		root := l.AppRoot()
-		for _, dir := range l.dirs {
-			dirs = append(dirs, fmt.Sprintf("%s/%s", root, dir))
-		}
-		return dirs
-	}()...)
-	return l.Load()
-}()
-
-// AppRoot returns the root directory of your application. UberFx developers
-// can edit this via the APP_ROOT environment variable. If the environment
-// variable is not set then it will fallback to the current working directory.
-func (l *Loader) AppRoot() string {
-	if appRoot, ok := l.lookUp(l.EnvironmentPrefix() + _appRoot); ok {
-		return appRoot
-	}
-
-	if cwd, err := os.Getwd(); err != nil {
-		panic(fmt.Sprintf("Unable to get the current working directory: %q", err.Error()))
-	} else {
-		return cwd
-	}
-}
-
-// ResolvePath returns an absolute path derived from AppRoot and the relative path.
-// If the input parameter is already an absolute path it will be returned immediately.
-func (l *Loader) ResolvePath(relative string) (string, error) {
-	if filepath.IsAbs(relative) {
-		return relative, nil
-	}
-
-	abs := path.Join(l.AppRoot(), relative)
-	if _, err := os.Stat(abs); err != nil {
-		return "", err
-	}
+// LookupFunc is a type alias for a function to look for environment variables,
+type LookUpFunc func(string) (string, bool)
 
-	return abs, nil
+// FileInfo represents a file to load.
+type FileInfo struct {
+	Name        string
+	Interpolate bool
 }
 
-func (l *Loader) baseFiles() []string {
-	// Order is important: last files override values in the first files.
-	return []string{_baseFile, l.Environment() + ".yaml"}
-}
-
-func (l *Loader) getResolver() FileResolver {
-	return NewRelativeResolver(l.Paths()...)
-}
+func LoadDefaults() (Provider, error) {
+	p, err := LoadFromFiles(
+		[]string{"${CONFIG_DIR:config}"},
+		[]FileInfo{
+			{Name: "base.yaml", Interpolate: true},
+			{Name: "${ENVIRONMENT:development}.yaml", Interpolate: true},
```

Comment:
Well, you don't have to do lookups yourself, e.g. environment will be set for you.
I can make it simple though.